chain/stmgr/utils.go

```diff
@@ -198,17 +220,34 @@ func GetSectorsForWinningPoSt(ctx context.Context, pv ffiwrapper.Verifier, sm *S
 		return nil, xerrors.Errorf("getting miner ID: %w", err)
 	}
 
-	ids, err := pv.GenerateWinningPoStSectorChallenge(ctx, wpt, abi.ActorID(mid), rand, uint64(len(sectorSet)))
+	ids, err := pv.GenerateWinningPoStSectorChallenge(ctx, wpt, abi.ActorID(mid), rand, numProvSect)
 	if err != nil {
 		return nil, xerrors.Errorf("generating winning post challenges: %w", err)
 	}
 
+	sectors, err := provingSectors.All(miner.SectorsMax)
```

Comment:
If you update go-bitfield, you should be able to do this without expanding the bitset.

```go
iter, err := provingSectors.BitIterator()
...
offset := 0
for i, n := range ids { // assuming the ids are sorted.
    sid, err := iter.Nth(n-offset)
    ...
    offset += n
}
```